spec/lib/search_query_transformer_spec.rb

```diff
@@ -68,13 +68,31 @@
     end
   end
 
-  context 'with \'before:"2022-01-01 23:00"\'' do
-    let(:query) { 'before:"2022-01-01 23:00"' }
+  context 'with \'is:"foo bar"\'' do
+    let(:query) { 'is:"foo bar"' }
 
     it 'transforms clauses' do
       expect(subject.send(:must_clauses)).to be_empty
       expect(subject.send(:must_not_clauses)).to be_empty
-      expect(subject.send(:filter_clauses).map(&:term)).to contain_exactly(lt: '2022-01-01 23:00', time_zone: 'UTC')
+      expect(subject.send(:filter_clauses).map(&:term)).to contain_exactly('foo bar')
+    end
+  end
+
+  context 'with \'before:"2022-01-01"\'' do
+    let(:query) { 'before:"2022-01-01"' }
+
+    it 'transforms clauses' do
+      expect(subject.send(:must_clauses)).to be_empty
+      expect(subject.send(:must_not_clauses)).to be_empty
+      expect(subject.send(:filter_clauses).map(&:term)).to contain_exactly(lt: '2022-01-01', time_zone: 'UTC')
+    end
+  end
+
+  context 'with \'before:"abc"\'' do
+    let(:query) { 'before:"abc"' }
+
+    it 'raises an exception' do
+      expect { subject }.to raise_error(Mastodon::FilterValidationError, 'Invalid date abc')
     end
   end
```

Comment:
Would it be worth using a shared example to test for each of the date operators? Not just testing only `before:`